lib/js/src/manager/screen/_AlertManagerBase.js

```diff
@@ -0,0 +1,226 @@
+        this._isAlertRpcAllowed = false;
+        this._fileManager = fileManager;
+        this._permissionManager = permissionManager;
+        this._nextCancelId = this._alertCancelIdMin;
+        this._softButtonObjects = [];
+
+        this._addListeners();
+        this._handleDisplayCapabilityUpdates();
+        this._handleTaskQueue();
+    }
+
+    /**
+     * Start the manager.
+     * @returns {Promise} - A promise.
+     */
+    async start () {
+        this._transitionToState(_SubManagerBase.READY);
+        await super.start();
+    }
+
+    /**
+     * Called when manager is being torn down
+     */
+    dispose () {
+        this._speechCapabilities = null;
+        this._isAlertRpcAllowed = false;
+        this._softButtonObjects = null;
+
+        this._handleTaskQueue();
+
+        if (this._permissionManager !== null) {
+            this._permissionManager.removeListener(this._permissionListener);
+        }
+
+        this._lifecycleManager.removeRpcListener(FunctionID.OnButtonPress, this._onButtonPressListener.bind(this));
+        this._lifecycleManager.removeRpcListener(FunctionID.OnButtonEvent, this._onButtonEventListener.bind(this));
+
+        super.dispose();
+    }
+
+    /**
+     * Presents an AlertView in the window.
+     * @param {AlertView} alert - An instance of AlertView to be displayed in the window.
+     */
+    async presentAlert (alert) {
+        if (this._getState() === _SubManagerBase.ERROR) {
+            console.log('AlertManager In Error State');
+        }
+
+        if (alert.getSoftButtons() !== null && alert.getSoftButtons !== undefined) {
+            if (!_ScreenManagerBase._checkAndAssignButtonIds(alert.getSoftButtons(), _ScreenManagerBase._ManagerLocation.ALERT_MANAGER)) {
+                console.log('Attempted to set soft button objects for Alert, but multiple buttons had the same id.');
+                return;
+            }
+
+            let softButtonObjects;
+            if (Array.isArray(alert.getSoftButtons())) {
+                softButtonObjects = alert.getSoftButtons();
+            } else {
+                softButtonObjects = [];
+            }
+
+            if (this._softButtonObjects.length > 0) {
+                for (const softButtonObject of softButtonObjects) {
+                    if (this._softButtonObjects.includes(softButtonObject)) {
+                        break;
+                    }
+                    this._softButtonObjects.push(softButtonObject);
+                }
+            } else {
+                this._softButtonObjects = softButtonObjects;
+            }
+        }
+
+        if (this._nextCancelId >= this._alertCancelIdMax) {
+            this._nextCancelId = this._alertCancelIdMin;
+        } else {
+            this._nextCancelId++;
+        }
+        const operation = new _PresentAlertOperation(this._lifecycleManager, alert, this._defaultMainWindowCapability, this._speechCapabilities, this._fileManager, this._nextCancelId++, this._listener);
```

Comment:
`this._listener` is never defined. The correct API for the presentAlert method should have a listener passed into it which is then passed into the operation class.